src/gui/hud.py

```diff
@@ -0,0 +1,72 @@
+from src.resources.font_rsc import *
+from src.helpers.text import *
+from src.gui.indented_rect import *
+
+
+class Hud:
+    SIDE_MARGIN = (LayoutRsc.WINDOW_WIDTH - LayoutRsc.GAME_AREA_WIDTH) / 2
+    TOP_MARGIN = 20
+
+    BORDER_WIDTH = LayoutRsc.LINE_THICKNESS
+    USABLE_AREA_WIDTH = LayoutRsc.GAME_AREA_WIDTH + BORDER_WIDTH * 2
+
+    TOP_BAR_HEIGHT = 50
+    TOP_BAR_INNER_MARGIN = 10
+    HEART_SPACING = 5
+    HEART_TEXTURE_SIZE = (29, 25)
+    HEART_TEXTURE_PATH = LayoutRsc.TEXTURES_PATH + 'hud/heart.png'
+
+    GAME_RECT_HEIGHT = LayoutRsc.GAME_AREA_HEIGHT + BORDER_WIDTH * 2
+
+    def __init__(self, game):
+        self._game = game
+        self._lives = 3
+        self._points = 0
+        self._top_bar_rect = IndentedRect(pos_x=Hud.SIDE_MARGIN,
+                                          pos_y=Hud.TOP_MARGIN,
+                                          width=Hud.USABLE_AREA_WIDTH,
+                                          height=Hud.TOP_BAR_HEIGHT)
+        self._game_rect = IndentedRect(pos_x=Hud.SIDE_MARGIN,
+                                       pos_y=Hud.TOP_MARGIN + self._top_bar_rect.get_height() + Hud.TOP_MARGIN,
+                                       width=Hud.USABLE_AREA_WIDTH,
+                                       height=Hud.GAME_RECT_HEIGHT)
+        self._heart_texture = pygame.transform.scale(pygame.image.load(Hud.HEART_TEXTURE_PATH), Hud.HEART_TEXTURE_SIZE)
+        self._game_surface = pygame.Surface((LayoutRsc.GAME_AREA_WIDTH, LayoutRsc.GAME_AREA_HEIGHT),
+                                            pygame.SRCALPHA, 32)
+
+    def is_running(self):
+        return self._game.is_running()
+
+    def process_events(self, joystick):
+        self._game.process_events(joystick)
+
+    def update(self, time_per_frame):
```

Comment:
No, it has to be time per frame. You need to know how much time elapsed since  last update so you can calculate all the positions. According to formula s =v*t. However, I can see that I mistakenly changed its name when refactoring and it's very confusing now. It should be delta_time. I will change it